doc_source/faces-detect-images.md

```diff
@@ -350,6 +350,92 @@ You can provide the input image as an image byte array \(base64\-encoded image b
       });
    ```
 
+------
+#### [ GO ]
+
+   This example displays the estimated age range , position , emotion and Gender for each detected faces\. Change the value of `photo` to the image file name\. Change the value of `bucket` to the Amazon S3 bucket where the image is stored\.
+
+   ```
+   //Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   //PDX-License-Identifier: MIT-0 (For details, see https://github.com/awsdocs/amazon-rekognition-developer-guide/blob/master/LICENSE-SAMPLECODE.)
+
+   package main
+
+    import (
+        "fmt"
+
+        "github.com/aws/aws-sdk-go/aws"
+        "github.com/aws/aws-sdk-go/aws/session"
+        "github.com/aws/aws-sdk-go/service/rekognition"
+    )
+
+    var svc *rekognition.Rekognition
+
+    func init() {
+
+        //https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/sessions.html
+
+        //Access keys are read from ~/.aws/credentials
+        sess, err := session.NewSession(&aws.Config{
+            Region: aws.String("us-east-1"),
+        })
+
+        if err != nil {
+            fmt.Println("Error while creating session,", err)
+            return
+        }
+
+        svc = rekognition.New(sess)
+        _ = svc
+    }
+
+    func main() {
+
```

Comment:
We're moving away from user's having to modify the code and using command-line flags:

bucket := flag.String("b", "", "The name of the bucket")
photo := flag.String("p", "", "The path to the photo file (JPEG, GIF, or ???")
flag.Parse()

if *bucket == "" || *photo == "" {
	fmt.Println("You must supply a bucket name (-b BUCKET) and photo file (-p PHOTO")
	return
}

Reply:
Done. I have added the command-line parsing to init()